--- misc/config.h.orig	2021-01-25 23:06:22.266896000 -0500
+++ misc/config.h	2021-01-25 23:31:45.451504000 -0500
@@ -494,7 +494,7 @@
    indirectly as a result of any use of cryptlib with this change made.  If
    you receive the code with the safety features already disabled, you must
    immediately obtain and use an original, unmodified version */
-/* #define USE_PKCS12 */
+#define USE_PKCS12
 /* Going beyond the PKCS #12 read capability which exists solely to allow 
    the import of keys supplied in that format by third parties, cryptlib has
    a PKCS #12 write capability.  This exists purely to allow the export of
@@ -509,7 +509,7 @@
    PKCS #12 read capability, as well as the fact that PKCS #12 write is an 
    unsupported facility with special-case usage restrictions that doesn't 
    work like any normal keyset */
-/* #define USE_PKCS12_WRITE */
+#define USE_PKCS12_WRITE
 
 #define USE_PKCS15
 #define USE_PGPKEYS
