--- session/ssh2_id.c.orig	2024-05-09 02:26:06.000000000 -0400
+++ session/ssh2_id.c	2026-03-14 09:09:12.196423000 -0400
@@ -404,7 +404,7 @@
 	assert( isWritePtr( sessionInfoPtr, sizeof( SESSION_INFO ) ) );
 	assert( isReadPtr( versionString, sizeof( versionStringLength ) ) );
 
-	REQUIRES( isShortIntegerRangeMin( versionStringLength, 3 ) );
+	REQUIRES( isShortIntegerRangeMin( versionStringLength, 2 ) );
 
 	if( ( versionStringLength >= 8 && \
 		  !memcmp( versionString, "AzureSSH", 8 ) ) )
@@ -919,7 +919,7 @@
 		versionStringLength = length - startOffset;
 		}
 	if( cryptStatusError( status ) || \
-		!isShortIntegerRangeMin( versionStringLength, 3 ) )
+		!isShortIntegerRangeMin( versionStringLength, 2 ) )
 		{
 		/* We need at least "-x.y" after the initial ID string, we can't 
 		   require any more than this because of CuteFTP (see note below).
@@ -933,7 +933,7 @@
 				  sanitiseString( sessionInfoPtr->receiveBuffer, 
 								  CRYPT_MAX_TEXTSIZE, length ) ) );
 		}
-	ENSURES( versionStringLength >= 3 && \
+	ENSURES( versionStringLength >= 2 && \
 			 versionStringLength < SSH_ID_MAX_SIZE );	/* From earlier checks */
 
 	/* Check whether the peer is using cryptlib */
