diff -ur ../cl-old/random/unix.c ./random/unix.c
--- ../cl-old/random/unix.c	2021-03-29 22:36:37.000000000 -0700
+++ ./random/unix.c	2021-03-29 22:38:03.000000000 -0700
@@ -505,7 +505,9 @@
 									   amount of output so typically gets
 									   truncated at SYSCTL_BUFFER_SIZE */
 #endif /* KERN_PROC2 */
+#ifdef GPROF_COUNT
 	{ 3, { CTL_KERN, KERN_PROF, GPROF_COUNT }, 10 },
+#endif
 									/* If kernel is compiled for profiling, 
 									   an array of statistical program 
 									   counter counts.  This typically isn't
