This patch doesn't solve the issue.
--- session/session.c.orig	2023-12-19 14:27:12.836848000 -0500
+++ session/session.c	2023-12-19 14:29:00.324330000 -0500
@@ -280,14 +280,19 @@
 	/* Check whether the certificate is valid at a standard level of 
 	   compliance, which catches expired certificates and other obvious
 	   problems */
+	// Don't mess with the cert!  If the compliance level is crap, do crap checks.
+#if 0
 	krnlSendMessage( iServerKey, IMESSAGE_SETATTRIBUTE, 
 					 ( MESSAGE_CAST ) &complianceLevelStandard, 
 					 CRYPT_OPTION_CERT_COMPLIANCELEVEL );
+#endif
 	status = krnlSendMessage( iServerKey, IMESSAGE_CHECK, NULL, 
 							  MESSAGE_CHECK_CERT );
+#if 0
 	krnlSendMessage( iServerKey, IMESSAGE_SETATTRIBUTE, 
 					 ( MESSAGE_CAST ) &complianceLevel, 
 					 CRYPT_OPTION_CERT_COMPLIANCELEVEL );
+#endif
 	if( cryptStatusOK( status ) )
 		return( CRYPT_OK );
 
