--- ./session/ssh2_channel.c.orig	2024-10-31 00:50:23.103563000 -0400
+++ ./session/ssh2_channel.c	2024-10-31 00:50:23.443226000 -0400
@@ -660,7 +660,7 @@
 		if( channelInfoPtr == NULL )
 			return( CRYPT_ERROR_NOTFOUND );
 		return( selectChannel( sessionInfoPtr, channelInfoPtr->writeChannelNo,
-							   CHANNEL_WRITE ) );
+							   CHANNEL_BOTH ) );
 		}
 	channelInfoPtr = ( SSH_CHANNEL_INFO * ) \
 				getCurrentChannelInfo( sessionInfoPtr, CHANNEL_READ );
