diff --git misc/os_spec.c misc/os_spec.c
index 12c3964..4dd7e1f 100644
--- misc/os_spec.c
+++ misc/os_spec.c
@@ -1300,8 +1300,8 @@ HMODULE WINAPI SafeLoadLibrary( IN_STRING LPCTSTR lpFileName )
 
 /* The size of the buffer for ACLs and the user token */
 
-#define ACL_BUFFER_SIZE		1024
-#define TOKEN_BUFFER_SIZE	256
+#define ACL_BUFFER_SIZE		4096
+#define TOKEN_BUFFER_SIZE	1024
 
 /* A composite structure to contain the various ACL structures.  This is
    required because ACL handling is a complex, multistage operation that
