--- misc/os_spec.h.orig	2018-01-10 01:39:04.000000000 -0500
+++ misc/os_spec.h	2018-01-29 18:20:02.023299000 -0500
@@ -223,10 +223,11 @@
      preprocessors get confused if they aren't defined */
   #define VC_16BIT( version )		0
   #define VC_LE_VC6( version )		0
-  #define VC_GE_2002( version )		0
-  #define VC_LT_2005( version )		0
+  #define VC_GE_2002( version )		1
+  #define VC_LT_2005( version )		1
   #define VC_GE_2005( version )		0
   #define VC_GE_2008( version )		0
+  #define VC_LT_2010( version )		1
   #define VC_GE_2010( version )		0
   #define VC_GE_2012( version )		0
   #define VC_GE_2013( version )		0
@@ -947,6 +948,9 @@
 	#include <machine/endian.h>
   #elif defined( __NetBSD__ )
 	#include <sys/endian.h>
+  #elif defined( __FreeBSD__ )
+	#include <sys/endian.h>
+  #elif defined(__MINGW32__)
   #else
 	#include <endian.h>
   #endif /* Apple vs. everyone else */
