--- misc/os_detect.h.orig	2023-12-31 10:15:40.782951000 -0500
+++ misc/os_detect.h	2023-12-31 10:16:40.940469000 -0500
@@ -658,7 +658,9 @@
 	 __GNUC__ is defined but the gcc include files aren't present.  The
 	 above checks catch the most common cases, if there are other pretend-
 	 gcc's then they'll need to be special-cased before this one */
+#if !defined(__MINGW32__)
   #include <endian.h>
+#endif
 #endif /* System-specific endian.h includes */
 
 #if defined( CONFIG_DATA_LITTLEENDIAN ) || defined( CONFIG_DATA_BIGENDIAN )
