--- session/ssl_hs.c.orig	2018-11-14 23:22:26.000000000 -0500
+++ session/ssl_hs.c	2020-01-23 18:23:41.236235000 -0500
@@ -240,6 +240,7 @@
 	const CIPHERSUITE_INFO **cipherSuiteInfo;
 	const BOOLEAN isServer = isServer( sessionInfoPtr ) ? TRUE : FALSE;
 	BOOLEAN allowDH = algoAvailable( CRYPT_ALGO_DH ) ? TRUE : FALSE;
+	BOOLEAN allowECCAuth = TRUE;
 	BOOLEAN allowECC = ( algoAvailable( CRYPT_ALGO_ECDH ) && \
 						 algoAvailable( CRYPT_ALGO_ECDSA ) ) ? TRUE : FALSE;
 	BOOLEAN allowRSA = algoAvailable( CRYPT_ALGO_RSA ) ? TRUE : FALSE;
@@ -268,7 +269,7 @@
 			{
 			/* There's no server private key present, we're limited to PSK
 			   suites */
-			allowECC = allowRSA = FALSE;
+			allowECC = allowRSA = allowECCAuth = FALSE;
 			}
 		else
 			{
@@ -278,7 +279,7 @@
 			   capable */
 			if( !checkContextCapability( sessionInfoPtr->privateKey,
 										 MESSAGE_CHECK_PKC_SIGN ) )
-				allowDH = allowECC = FALSE;
+				allowDH = allowECC = allowECCAuth = FALSE;
 
 			/* To be usable for ECC or RSA the server key has to itself be 
 			   an ECC or RSA key */
@@ -286,13 +287,16 @@
 									  IMESSAGE_GETATTRIBUTE, &pkcAlgo,
 									  CRYPT_CTXINFO_ALGO );
 			if( cryptStatusError( status ) )
-				allowECC = allowRSA = FALSE;
+				allowECC = allowRSA = allowECCAuth = FALSE;
 			else
 				{
 				if( !isEccAlgo( pkcAlgo ) )
 					allowECC = FALSE;
 				if( pkcAlgo != CRYPT_ALGO_RSA )
+					{
 					allowRSA = FALSE;
+					allowECCAuth = FALSE;
+					}
 				}
 			}
 		}
@@ -443,8 +447,13 @@
 			( cipherSuiteInfoPtr->flags & CIPHERSUITE_FLAG_DH ) )
 			continue;
 		if( !allowECC && \
-			( cipherSuiteInfoPtr->flags & CIPHERSUITE_FLAG_ECC ) )
+			( cipherSuiteInfoPtr->flags & CIPHERSUITE_FLAG_ECC ) && \
+			( cipherSuiteInfoPtr->authAlgo != CRYPT_ALGO_RSA) )
 			continue;
+		if( !allowECCAuth && \
+			( cipherSuiteInfoPtr->flags & CIPHERSUITE_FLAG_ECC ) && \
+			( cipherSuiteInfoPtr->authAlgo == CRYPT_ALGO_RSA) )
+			continue;
 		if( !allowTLS12 && \
 			( cipherSuiteInfoPtr->flags & CIPHERSUITE_FLAG_TLS12 ) )
 			continue;
@@ -521,7 +530,7 @@
 	   find out that we can use it */
 	if( altSuiteIndex < cipherSuiteInfoSize )
 		{
-		REQUIRES( allowECC );
+		REQUIRES( allowECCAuth );
 
 		handshakeInfo->eccSuiteInfoPtr = cipherSuiteInfo[ altSuiteIndex ];
 		}
