--- test/test.h.orig	2023-12-31 07:11:19.119452000 -0500
+++ test/test.h	2023-12-31 07:11:41.022395000 -0500
@@ -378,6 +378,7 @@
   #define THREAD_SLEEP( ms ) Sleep( ms )
   typedef unsigned ( __stdcall *THREAD_FUNC )( void *arg );
 #elif defined( UNIX_THREADS )
+  #include <sys/select.h>
   #define THREAD_HANDLE		pthread_t
   #define THREAD_EXIT()		pthread_exit( ( void * ) 0 )
   #define THREAD_SELF()		pthread_self()
