--- session/ssl_ext.c.orig	2020-01-23 15:25:30.640683000 -0500
+++ session/ssl_ext.c	2020-01-23 15:25:40.141297000 -0500
@@ -607,8 +607,8 @@
 #endif /* CONFIG_SUITEB */
 
 		/* Make sure that the curve matches the server's signing key */
-		if( curveSize != keySize )
-			continue;
+		//if( curveSize != keySize )
+		//	continue;
 
 		/* We've got a matching curve, remember it.  In theory we could exit
 		   at this point but we continue anyway to clear the remainder of 
