--- ../tmp2/misc/os_detect.h	2019-01-31 14:57:46.000000000 -0500
+++ misc/os_detect.h	2019-06-03 18:35:55.358499000 -0400
@@ -241,10 +241,11 @@
      preprocessors get confused if they aren't defined */
   #define VC_16BIT( version )		0
   #define VC_LE_VC6( version )		0
-  #define VC_GE_2002( version )		0
-  #define VC_LT_2005( version )		0
+  #define VC_GE_2002( version )		1
+  #define VC_LT_2005( version )		1
   #define VC_GE_2005( version )		0
   #define VC_GE_2008( version )		0
+  #define VC_LT_2010( version )		1
   #define VC_GE_2010( version )		0
   #define VC_GE_2012( version )		0
   #define VC_GE_2013( version )		0
