--- tools/getlibs.sh.orig	2021-01-24 07:40:21.569115000 -0500
+++ tools/getlibs.sh	2021-01-24 07:40:42.823333000 -0500
@@ -59,9 +59,9 @@
 
 if hasSubstring "$BUILDOPTS" "sanitize=safe-stack" ; then
 	CLANG_VER="$(clang -dumpversion | tr -d  '.' | cut -c 1-2)" ;
-	if [ $CLANG_VER -gt 47 ] ; then
-		LDARGS="$LDARGS -fsanitize=safe-stack" ;
-	fi ;
+	#if [ $CLANG_VER -gt 47 ] ; then
+	#	LDARGS="$LDARGS -fsanitize=safe-stack" ;
+	#fi ;
 fi
 
 # Add any libraries needed by optional components.  In the case of zlib use
--- tools/ccopts.sh.orig	2020-11-26 02:40:05.222021000 -0500
+++ tools/ccopts.sh	2021-01-24 07:41:02.126230000 -0500
@@ -358,9 +358,9 @@
 # tools/getlibs.sh also for clang 4.7 or newer.
 
 if [ $ISCLANG -gt 0 ] && [ $ISSPECIAL -eq 0 ] ; then
 	CLANG_VER="$($CC -dumpversion | sed -E 's/^([0-9]+)$/\1.0/' | sed -E 's/^([0-9]+)\.([0-9]).*$/\1\2/')" ;
-	if [ $CLANG_VER -gt 42 ] ; then
-		CCARGS="$CCARGS -fsanitize=safe-stack" ;
-	fi ;
+	#if [ $CLANG_VER -gt 42 ] ; then
+	#	CCARGS="$CCARGS -fsanitize=safe-stack" ;
+	#fi ;
 fi
 
 # The Sun compiler has its own set of problems, the biggest of which is
