--- tools/ccopts.sh.orig	2019-06-20 12:51:38.118150000 -0400
+++ tools/ccopts.sh	2019-06-20 12:52:34.479052000 -0400
@@ -210,7 +210,8 @@
 	for includepath in $ODBCPATHS ; do
 		if [ -f $includepath ] ; then
 			echo "ODBC interface detected, enabling ODBC support." >&2 ;
-			CCARGS="$CCARGS -DHAS_ODBC -I"$(dirname $includepath)"" ;
+			#CCARGS="$CCARGS -DHAS_ODBC -I"$(dirname $includepath)"" ;
+			echo "Hah ha ha, no. fuck off." >&2 ;
 			break ;
 		fi
 	done
