--- ./crypt/aesopt.h.orig	2023-07-11 19:08:32.000000000 -0400
+++ ./crypt/aesopt.h	2024-10-31 00:50:22.127051000 -0400
@@ -106,7 +106,7 @@
 	defined( _M_IX86 ) && \
 	!( defined( _WIN32_WCE ) || defined( NO_ASM ) )
   #ifndef USE_VIA_ACE_IF_PRESENT
-	#define USE_VIA_ACE_IF_PRESENT
+//	#define USE_VIA_ACE_IF_PRESENT
   #endif
 #endif /* VC++ on x86 under Win32 - pcg */
 
@@ -216,7 +216,7 @@
 #if defined( __GNUC__ ) && defined( __i386__ ) \
  || defined( _WIN32 ) && defined( _M_IX86 ) && !(defined( _WIN64 ) \
  || defined( _WIN32_WCE ) || defined( _MSC_VER ) && ( _MSC_VER <= 800 ))
-#  define VIA_ACE_POSSIBLE
+//#  define VIA_ACE_POSSIBLE
 #endif
 
 /* AES is supported out of the box by Windows x64 compilers, but by gcc only 
@@ -285,7 +285,7 @@
 	AES_REV_DKS must be set for assembler code used with a VIA ACE build
 */
 
-#if 1 && defined( VIA_ACE_POSSIBLE ) && !defined( USE_VIA_ACE_IF_PRESENT )
+#if 0 && defined( VIA_ACE_POSSIBLE ) && !defined( USE_VIA_ACE_IF_PRESENT )
 #  define USE_VIA_ACE_IF_PRESENT
 #endif
 
@@ -530,7 +530,7 @@
 #endif
 
 #if defined( ASSUME_VIA_ACE_PRESENT ) && !defined( USE_VIA_ACE_IF_PRESENT )
-#  define USE_VIA_ACE_IF_PRESENT
+//#  define USE_VIA_ACE_IF_PRESENT
 #endif
 
 #if defined( __APPLE__ ) && TARGET_OS_SIMULATOR
