--- ./makefile.orig	2024-10-31 00:50:22.293145000 -0400
+++ ./makefile	2024-10-31 00:50:22.799649000 -0400
@@ -1815,7 +1815,7 @@
 # than the system tools and libraries for the build, so we special-case this
 # step based on the $(OSNAME) setting supplied to the build script.
 
-$(ALIBNAME):	$(OBJS) $(EXTRAOBJS) $(TESTOBJS)
+$(ALIBNAME):	$(OBJS) $(EXTRAOBJS)
 				@./tools/buildlib.sh $(ALIBNAME) $(OSNAME) $(AR) \
 					$(OBJS) $(EXTRAOBJS)
 
