--- session/tls_suites.c.orig	2023-02-28 02:55:06.000000000 -0500
+++ session/tls_suites.c	2023-12-31 07:30:02.368446000 -0500
@@ -414,12 +414,12 @@
 
 	/* ECC suites if these are preferred */
 #ifdef PREFER_ECC
-  #ifdef USE_GCM
-	{ cipherSuiteGCM, FAILSAFE_ARRAYSIZE( cipherSuiteGCM, CIPHERSUITE_INFO ) },
-  #endif /* USE_GCM */
   #if defined( USE_ECDH )
 	{ cipherSuiteECC, FAILSAFE_ARRAYSIZE( cipherSuiteECC, CIPHERSUITE_INFO ) },
   #endif /* USE_ECDH */
+  #ifdef USE_GCM
+	{ cipherSuiteGCM, FAILSAFE_ARRAYSIZE( cipherSuiteGCM, CIPHERSUITE_INFO ) },
+  #endif /* USE_GCM */
   #ifdef USE_CHACHA20
 	{ cipherSuiteBernstein, FAILSAFE_ARRAYSIZE( cipherSuiteBernstein, CIPHERSUITE_INFO ) },
   #endif /* USE_CHACH20 */
