--- ./session/ssh2_authcli.c.orig	2024-10-31 00:50:22.587803000 -0400
+++ ./session/ssh2_authcli.c	2024-10-31 00:50:23.003964000 -0400
@@ -597,6 +597,7 @@
 	   and return some sort of useful information to the caller */
 	if( providedAuthType == SSH_AUTHTYPE_PUBKEY )
 		{
+#if 0
 		if( needsPW )
 			{
 			setObjectErrorInfo( sessionInfoPtr, CRYPT_SESSINFO_PASSWORD,
@@ -606,6 +607,7 @@
 					  "Server requested password authentication but only a "
 					  "public/private key was available" ) );
 			}
+#endif
 		return CRYPT_ENVELOPE_RESOURCE;
 		}
 	if( requiredAuthType == SSH_AUTHTYPE_PUBKEY )
