--- kernel/thread.h.orig	2023-12-13 12:08:02.174014000 -0500
+++ kernel/thread.h	2023-12-13 12:08:24.634971000 -0500
@@ -3013,10 +3013,7 @@
 	 underlying process context should yield the associated thread */
   #define THREAD_YIELD()		yield()
 #else
-  #if defined( __linux__ ) && !defined( __USE_GNU )
-	void pthread_yield( void );
-  #endif /* Present but not prototyped unless GNU extensions are enabled */
-  #define  THREAD_YIELD()		pthread_yield()
+  #define  THREAD_YIELD()		sched_yield()
 #endif /* Not-very-portable Posix portability */
 #define THREAD_SLEEP( ms )		{ \
 								struct timeval tv = { 0 }; \
