--- ../cl/session/ssh2_authc.c	2012-09-10 22:40:52.000000000 -0700
+++ session/ssh2_authc.c	2015-03-20 21:23:03.000000000 -0700
@@ -529,7 +529,7 @@
 		if( !cryptStatusError( status ) )
 			{
 			status = CRYPT_OK;	/* readUint32() returns a count value */
-			if( noPrompts <= 0 || noPrompts > 4 )
+			if( noPrompts < 0 || noPrompts > 4 )
 				{
 				/* Requesting zero or more than a small number of prompts is 
 				   suspicious */
@@ -537,37 +537,39 @@
 				}
 			}
 		}
-	if( cryptStatusOK( status ) )
-		status = readString32( &stream, promptBuffer, 
-							   CRYPT_MAX_TEXTSIZE, &promptLength );
-	sMemDisconnect( &stream );
-	if( cryptStatusError( status ) )
-		{
-		retExt( status,
-				( status, SESSION_ERRINFO, 
-				  "Invalid PAM authentication request packet" ) );
-		}
+	if ( noPrompts > 0 ) {
+		if( cryptStatusOK( status ) )
+			status = readString32( &stream, promptBuffer, 
+								   CRYPT_MAX_TEXTSIZE, &promptLength );
+		sMemDisconnect( &stream );
+		if( cryptStatusError( status ) )
+			{
+			retExt( status,
+					( status, SESSION_ERRINFO, 
+					  "Invalid PAM authentication request packet" ) );
+			}
 
-	/* Make sure that we're being asked for some form of password 
-	   authentication.  This assumes that the prompt string begins with the 
-	   word "password" (which always seems to be the case), if it isn't then 
-	   it may be necessary to do a substring search */
-	if( promptLength < 8 || strCompare( promptBuffer, "Password", 8 ) )
-		{
+		/* Make sure that we're being asked for some form of password 
+		   authentication.  This assumes that the prompt string begins with the 
+		   word "password" (which always seems to be the case), if it isn't then 
+		   it may be necessary to do a substring search */
+		if( promptLength < 8 || strCompare( promptBuffer, "Password", 8 ) )
+			{
 		/* The following may produce somewhat inconsistent results in terms
 		   of what it reports because it's unclear what 'name' actually is, 
 		   on the off chance that something fills this in it could produce
 		   a less appropriate error message than the prompt, but we 
 		   opportunistically try it in case it contains something useful */
-		retExt( CRYPT_ERROR_BADDATA,
-				( CRYPT_ERROR_BADDATA, SESSION_ERRINFO, 
-				  "Server requested unknown PAM authentication type '%s'", 
-				  ( nameLength > 0 ) ? \
-				  sanitiseString( nameBuffer, CRYPT_MAX_TEXTSIZE, \
-								  nameLength ) : \
-				  sanitiseString( promptBuffer, CRYPT_MAX_TEXTSIZE, \
-								  promptLength ) ) );
-		}
+			retExt( CRYPT_ERROR_BADDATA,
+					( CRYPT_ERROR_BADDATA, SESSION_ERRINFO, 
+					  "Server requested unknown PAM authentication type '%s'", 
+					  ( nameLength > 0 ) ? \
+					  sanitiseString( nameBuffer, CRYPT_MAX_TEXTSIZE, \
+									  nameLength ) : \
+					  sanitiseString( promptBuffer, CRYPT_MAX_TEXTSIZE, \
+									  promptLength ) ) );
+			}
+	}
 
 	REQUIRES( passwordPtr != NULL && \
 			  passwordPtr->valueLength > 0 && \
