--- io/tcp.c.orig	2018-03-12 13:47:37.385998000 -0400
+++ io/tcp.c	2018-03-12 13:48:12.696032000 -0400
@@ -402,7 +402,7 @@
 		return( status );
 	LOOP_MED( ( selectIterations = 0, status = SOCKET_ERROR ), \
 			  isSocketError( status ) && \
-				!checkMonoTimerExpired( &timerInfo ) && \
+				(selectIterations == 0 || !checkMonoTimerExpired( &timerInfo )) && \
 				selectIterations < 20, 
 			  selectIterations++ )
 		{
