--- session/ssh2_crypt.c.orig	2019-01-21 16:56:04.000000000 -0500
+++ session/ssh2_crypt.c	2020-02-14 02:36:21.428805000 -0500
@@ -55,12 +55,12 @@
 	/* Walk along the counter incrementing each byte if required */
 	LOOP_MED( i = blockSize - 1, i >= 0, i-- )
 		{
-		if( ctrPtr[ i ]++ != 0 )
+		if( ++ctrPtr[ i ] != 0 )
 			break;
 		}
 	ENSURES( LOOP_BOUND_OK );
 	ENSURES( ctrMSB != ctrPtr[ blockSize - 1 ] );
-	ENSURES( ctrMSB != 0 || ctrMSBnext != ctrPtr[ blockSize - 2 ] );
+	ENSURES( (ctrPtr[ blockSize - 1 ] != 0 || ctrMSBnext != ctrPtr[ blockSize - 2 ]) );
 
 	return( CRYPT_OK );
 	}
