--- ./bn/bn_lcl.h.orig	2024-05-25 02:14:06.000000000 -0400
+++ ./bn/bn_lcl.h	2024-10-31 00:50:22.858004000 -0400
@@ -232,7 +232,7 @@
 #  define PTR_SIZE_INT size_t
 # endif                         /* defined(OPENSSL_SYS_VMS) [else] */
 
-# if !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_NO_INLINE_ASM) && !defined(PEDANTIC)
+# if !defined(PEDANTIC)
 
 /* The original OpenSSL code has a mass of inline asm to handle 64-bit 
    multiplies across different architectures, however if the compiler 
