--- test/certs.c.orig	2018-01-03 02:46:59.742613000 -0500
+++ test/certs.c	2018-01-03 02:47:14.150003000 -0500
@@ -17,7 +17,7 @@
   #define _OSSPEC_DEFINED
   #if !defined( __WIN32__ ) && ( defined( WIN32 ) || defined( _WIN32 ) )
 	#define __WIN32__
-  #elif !defined( __WIN64__ ) defined( _M_X64 )
+  #elif !defined( __WIN64__ ) && defined( _M_X64 )
 	#define __WIN64__
   #endif /* Win32/Win64 */
   #define VC_LT_2005( version )		( version < 1400 )
