--- cert/dn.c.orig	2018-02-15 01:13:14.949502000 -0500
+++ cert/dn.c	2018-02-15 01:13:55.474689000 -0500
@@ -204,7 +204,7 @@
   /*W*/				   xF|									  xS,					   /*W*/
   /*X*/	0,																			   /*X*/
   /*Y*/				xE|											 xT,				   /*Y*/
-  /*Z*/	xA|									xM|							  xW,		   /*Z*/
+  /*Z*/	xA|									xM|							  xW|	   xZ, /*Z*/
 	/*	 A  B  C  D  E  F  G  H  I  J  K  L  M  N  O  P  Q  R  S  T  U  V  W  X  Y  Z */
 		0, 0	/* Catch overflows */
 		};
