--- ../tmp2/tools/ccopts.sh	2019-03-04 16:32:32.000000000 -0500
+++ tools/ccopts.sh	2019-06-03 16:50:00.486529000 -0400
@@ -259,10 +259,10 @@
 	done
 
 fi
-if [ -f /usr/include/zlib.h ] ; then
-	echo "  (Enabling use of system zlib)." >&2 ;
-	CCARGS="$CCARGS -DHAS_ZLIB" ;
-fi
+#if [ -f /usr/include/zlib.h ] ; then
+#	echo "  (Enabling use of system zlib)." >&2 ;
+#	CCARGS="$CCARGS -DHAS_ZLIB" ;
+#fi
 
 # If we're building a development or analysis build, enable various unsafe
 # options that are normally disabled by default
