--- tools/ccopts.sh.orig	2010-12-19 00:57:12.000000000 -0800
+++ tools/ccopts.sh	2013-01-05 22:33:35.000000000 -0800
@@ -105,7 +105,7 @@
 			HASDYNLOAD=1 ;
 		fi ;;
 esac
-if [ $HASDYNLOAD -gt 0 ] ; then
+if [ $HASDYNLOAD -gt 0 -a 1 = 2 ] ; then
 	if [ -f /usr/include/sql.h ] ; then
 		echo "ODBC interface detected, enabling ODBC support." >&2 ;
 		CCARGS="$CCARGS -DHAS_ODBC" ;
@@ -176,7 +176,6 @@
 # of large-displacement jumps, so if you're tuning the code for size/speed
 # you can try -fpic to see if you get any improvement.
 
-if [ $# -eq 2 ] ; then
 	case $OSNAME in
 		'Darwin')
 			CCARGS="$CCARGS -fPIC -fno-common" ;;
@@ -197,7 +196,6 @@
 		*)
 			CCARGS="$CCARGS -fPIC" ;;
 	esac ;
-fi
 
 # Conversely, if we're building a static lib and the system requires it, set
 # up static lib-specific options.
@@ -448,7 +446,7 @@
 # later versions of gcc 4.1.x, to be on the safe side we only enable it
 # for gcc 4.2 and newer.
 
-if [ $GCC_VER -ge 42 ] ; then
+if [ $GCC_VER -ge 42 -a 1 = 3 ] ; then
   if [ `$CC -fstack-protector -S -o /dev/null -xc /dev/null 2>&1 | grep -c "unrecog"` -eq 0 ] ; then
 	CCARGS="$CCARGS -fstack-protector" ;
   fi ;
