--- js-1.8.5/js/src/configure.in.orig	2024-03-16 23:06:25
+++ js-1.8.5/js/src/configure.in	2024-03-16 23:06:47
@@ -196,8 +196,8 @@
     if test -z "$MIDL"; then MIDL=midl; fi
     ;;
 *-darwin*)
-    if test -z "$CC"; then CC=gcc-4.2; fi
-    if test -z "$CXX"; then CXX=g++-4.2; fi
+    if test -z "$CC"; then CC=cc; fi
+    if test -z "$CXX"; then CXX=c++; fi
     ;;
 esac
 fi
--- js-1.8.5/js/src/configure.orig	2024-03-20 01:29:39
+++ js-1.8.5/js/src/configure	2024-03-20 01:30:55
@@ -953,8 +953,8 @@
     if test -z "$MIDL"; then MIDL=midl; fi
     ;;
 *-darwin*)
-    if test -z "$CC"; then CC=gcc-4.2; fi
-    if test -z "$CXX"; then CXX=g++-4.2; fi
+    if test -z "$CC"; then CC=cc; fi
+    if test -z "$CXX"; then CXX=c++; fi
     ;;
 esac
 fi
@@ -3873,7 +3873,7 @@
     echo "$ac_t""yes" 1>&6    
 fi
 
-  for ac_prog in $PYTHON python2.7 python2.6 python2.5 python
+  for ac_prog in $PYTHON python2.7 python2.6 python2.5 python python3
 do
 # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
