--- js-1.8.5/js/src/configure.in.orig	2023-01-25 20:00:45.568773908 -0800
+++ js-1.8.5/js/src/configure.in	2023-01-25 19:59:37.694768104 -0800
@@ -1700,6 +1700,7 @@
     if test "$ac_has_werror_return_type" = "yes"; then
         _WARNINGS_CXXFLAGS="${_WARNINGS_CXXFLAGS} -Werror=return-type"
     fi
+    _WARNINGS_CXXFLAGS="-w"
 
 else
     _DEFINES_CXXFLAGS='-DMOZILLA_CLIENT -D_JS_CONFDEFS_H_ $(ACDEFINES)'
