diff -ur ../mozjs-pre/js-1.8.5/js/src/methodjit/MethodJIT.cpp ./js-1.8.5/js/src/methodjit/MethodJIT.cpp
--- ../mozjs-pre/js-1.8.5/js/src/methodjit/MethodJIT.cpp	2021-03-13 19:50:35.282449000 -0500
+++ ./js-1.8.5/js/src/methodjit/MethodJIT.cpp	2021-03-13 19:53:09.753422000 -0500
@@ -189,7 +189,7 @@
 JS_STATIC_ASSERT(JSVAL_TAG_MASK == 0xFFFF800000000000LL);
 JS_STATIC_ASSERT(JSVAL_PAYLOAD_MASK == 0x00007FFFFFFFFFFFLL);
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(JaegerTrampoline) "\n"
 SYMBOL_STRING(JaegerTrampoline) ":"       "\n"
@@ -236,7 +236,7 @@
     "jmp *0(%rsp)"                      "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(JaegerTrampolineReturn) "\n"
 SYMBOL_STRING(JaegerTrampolineReturn) ":"       "\n"
@@ -256,7 +256,7 @@
     "ret"                                "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(JaegerThrowpoline)  "\n"
 SYMBOL_STRING(JaegerThrowpoline) ":"        "\n"
@@ -281,7 +281,7 @@
 
 JS_STATIC_ASSERT(offsetof(VMFrame, regs.fp) == 0x38);
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(InjectJaegerReturn)   "\n"
 SYMBOL_STRING(InjectJaegerReturn) ":"         "\n"
@@ -309,7 +309,7 @@
 JS_STATIC_ASSERT(offsetof(VMFrame, savedEBX) == 0x2c);
 JS_STATIC_ASSERT(offsetof(VMFrame, regs.fp) == 0x1C);
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(JaegerTrampoline) "\n"
 SYMBOL_STRING(JaegerTrampoline) ":"       "\n"
@@ -339,7 +339,7 @@
     "jmp *16(%ebp)"                      "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(JaegerTrampolineReturn) "\n"
 SYMBOL_STRING(JaegerTrampolineReturn) ":" "\n"
@@ -357,7 +357,7 @@
     "ret"                                "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(JaegerThrowpoline)  "\n"
 SYMBOL_STRING(JaegerThrowpoline) ":"        "\n"
@@ -388,7 +388,7 @@
 
 JS_STATIC_ASSERT(offsetof(VMFrame, regs.fp) == 0x1C);
 
-asm volatile (
+asm (
 ".text\n"
 ".globl " SYMBOL_STRING(InjectJaegerReturn)   "\n"
 SYMBOL_STRING(InjectJaegerReturn) ":"         "\n"
@@ -423,7 +423,7 @@
 #define FUNCTION_HEADER_EXTRA
 #endif
 
-asm volatile (
+asm (
 ".text\n"
 FUNCTION_HEADER_EXTRA
 ".globl " SYMBOL_STRING(InjectJaegerReturn) "\n"
@@ -436,7 +436,7 @@
     "bx  lr"                                "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 FUNCTION_HEADER_EXTRA
 ".globl " SYMBOL_STRING(JaegerTrampoline)   "\n"
@@ -495,7 +495,7 @@
 "   bx     r4"                                  "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 FUNCTION_HEADER_EXTRA
 ".globl " SYMBOL_STRING(JaegerTrampolineReturn)   "\n"
@@ -515,7 +515,7 @@
 "   pop     {r4-r11,pc}"                    "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 FUNCTION_HEADER_EXTRA
 ".globl " SYMBOL_STRING(JaegerThrowpoline)  "\n"
@@ -540,7 +540,7 @@
 "   pop     {r4-r11,pc}"                    "\n"
 );
 
-asm volatile (
+asm (
 ".text\n"
 FUNCTION_HEADER_EXTRA
 ".globl " SYMBOL_STRING(JaegerStubVeneer)   "\n"
