--- session/sess_attr.c.orig	2017-07-27 17:58:38.000000000 -0400
+++ session/sess_attr.c	2018-01-03 01:44:45.682545000 -0500
@@ -103,11 +103,13 @@
 	   then we can't set a server name as well */
 	if( sessionInfoPtr->transportSession != CRYPT_ERROR )
 		return( exitErrorInited( sessionInfoPtr, CRYPT_SESSINFO_SESSION ) );
+/*
 	if( sessionInfoPtr->networkSocket != CRYPT_ERROR )
 		{
 		return( exitErrorInited( sessionInfoPtr, 
 								 CRYPT_SESSINFO_NETWORKSOCKET ) );
 		}
+*/
 
 	/* Parse the server name.  The PKI protocols all use HTTP as their 
 	   substrate so if it's not SSH or SSL/TLS we require HTTP */
